features/netsocket/WiFiInterface.h

```diff
@@ -100,6 +100,118 @@ class WiFiInterface: public NetworkInterface
      *                   negative on error see @a nsapi_error
      */
     virtual nsapi_size_or_error_t scan(WiFiAccessPoint *res, nsapi_size_t count) = 0;
+
+    /** WiFi Access Point API
+     *
+     *  WiFi Access Point functionality. Support for AP mode is not mandatory in which case AP mode API functions will
+     *  return @a NSAPI_ERROR_UNSUPPORTED or NULL.
+     */
+
+    /** Set IP config for access point
+     *
+     *  This function has to be called before the access point is started.
+     *
+     *  @param gateway     Null-terminated representation of the local gateway
+     *  @param netmask     Null-terminated representation of the network mask
+     *  @return            0 on success, negative error code on failure
+     */
+    virtual nsapi_error_t ap_set_network(const char *gateway, const char *netmask)
```

Comment:
I just couldn't find good use case for having getters for them as the data originates from the user anyway.